plugin.py

```diff
@@ -14,6 +40,53 @@ class RustAnalyzer(AbstractPlugin):
     def name(cls) -> str:
         return "rust-analyzer"
 
+    @classmethod
+    def basedir(cls) -> str:
+        return os.path.join(cls.storage_path(), "LSP-{}".format(cls.name()))
+
+    @classmethod
+    def server_version(cls) -> str:
+        settings = sublime.load_settings("LSP-rust-analyzer.sublime-settings")
+        return str(settings.get("version"))
+
+    @classmethod
+    def current_server_version(cls) -> str:
+        with open(os.path.join(cls.basedir(), "VERSION"), "r") as fp:
+            return fp.read()
+
+    @classmethod
+    def needs_update_or_installation(cls) -> bool:
+        try:
+            return cls.server_version() != cls.current_server_version()
+        except OSError:
+            return True
+
+    @classmethod
+    def install_or_update(cls) -> None:
+        try:
+            shutil.rmtree(cls.basedir(), ignore_errors=True)
```

Comment:
I feel it's better to do an explicit `path.isdir` check instead of using `ignore_errors=True` since we're gonna get more appropriate error if deleting fails (due to files being locked, for example).

Reply:
I added path.isdir